File: core/src/com/mygdx/game/handler/PassHandler.java
package com.mygdx.game.handler;

public class PassHandler extends BaseHandler {

    @Override
    public BaseHandler.HandlerEntity doHandle(BaseHandler.HandlerEntity s, HandlerChain chain) {
        int moveCount = s.getMoveCount();

        while (moveCount > 0) {
            doTask2(s);
            System.out.println("移动：" + moveCount);
            // 移动完毕的判断，路障或者银行
            moveCount--;
        }
        return chain.process(s);
    }

    private String doTask2(HandlerEntity entity) {
        ResultReporter<String> reporter = new ResultReporter<>();
        ReportUtils.deley(1, reporter, "walk");
        return reporter.waitReport();
    }

}

File: core/src/main/kotlin/com/rich/diy/game/handler/PassHandler.kt
package com.rich.diy.game.handler

import com.rich.diy.game.entity.WayPoint
import com.rich.diy.game.events.ActionEvent


class PassHandler : BaseHandler() {
    override fun doHandle(s: HandlerEntity, chain: HandlerChain): HandlerEntity? {
        var moveCount = s.moveCount
        while (moveCount > 0) {
            var wayPoint = doTask2(s)
            println("移动：$moveCount 到 $wayPoint")
            // 移动完毕的判断，路障或者银行
            moveCount--
        }
        return chain.process(s)
    }

    private fun doTask2(entity: HandlerEntity): WayPoint {
        val reporter: ResultReporter<WayPoint> = ResultReporter<WayPoint>()
//        MainGame.Instance.doWalk(entity.player!!, reporter)
        return ActionEvent<WayPoint>("walk").apply {
            handlerEntity = entity
        }.postAndWait()
    }
}